### PR TITLE
NECO-172 enabled to delete user metadata.

### PR DESCRIPTION
**Pull request by hishida**
## Ticket

[NECO-172](https://pflabo.atlassian.net/browse/NECO-172)
## Changes
- Implemented `remove_meta` method to enable to delete all/specified user metadata.
- Added corresponding tests.
## Tests
- Tested to success all rspec tests.
- Tested to delete all/specified user metadata.

Details are in [ticket](https://pflabo.atlassian.net/browse/NECO-172).

### CODE REVIEW

File: lib/ricohapi/mstorage/client.rb [hunk 1]
```diff
@@ -95,7 +95,18 @@ def add_meta(media_id, user_meta)
...
+        when 'user'
+          handle_response(:as_raw) do
+            token.delete endpoint_for("media/#{media_id}/meta/user")
+          end
+        when USER_META_REGEX
+          handle_response(:as_raw) do
+            token.delete endpoint_for("media/#{media_id}/meta/user/#{$1}")
+          end
+        else
+          raise Error.new("invalid key: #{key}")
```
Comment (sohda) [inline lib/ricohapi/mstorage/client.rb]: I saw your test result screenshot, I think following code is better: 

``` ruby
raise Error.new("invalid key: #{key.inspect}")
```

If it is the  above code, we can know the key is either nil or "".

Comment (hishida) [inline lib/ricohapi/mstorage/client.rb]: Thank you, that's good!
I'll fix as above.

File: lib/ricohapi/mstorage/client.rb [hunk 2]
```diff
@@ -95,7 +95,18 @@ def add_meta(media_id, user_meta)
...
+        when 'user'
+          handle_response(:as_raw) do
+            token.delete endpoint_for("media/#{media_id}/meta/user")
+          end
+        when USER_META_REGEX
+          handle_response(:as_raw) do
+            token.delete endpoint_for("media/#{media_id}/meta/user/#{$1}")
+          end
+        else
+          raise Error.new("invalid key: #{key.inspect}")
```
Comment (sohda) [inline lib/ricohapi/mstorage/client.rb:108]: I found the same code at here. > https://github.com/hishida/media-storage-rb/blob/cd8f4767ad4a114e940b57f1e56204249f49a113/lib/ricohapi/mstorage/client.rb#L73
What are we gonna do?

Comment (hishida) [inline lib/ricohapi/mstorage/client.rb:108]: Thank you for your noticing.
I'll fix L73 similarly.

**Comment by sohda** [comment]
Thank you for coding.
I'm gonna merge this pull request!

**Pull request merged**